secure/storage/src/vault.rs

```diff
@@ -209,38 +128,12 @@ impl VaultStorage {
             .version)
     }
 
-    pub fn set_policies(
-        &self,
-        name: &str,
-        engine: &VaultEngine,
-        policy: &Policy,
-    ) -> Result<(), Error> {
-        for perm in &policy.permissions {
-            match &perm.id {
-                Identity::User(id) => self.set_policy(id, engine, name, &perm.capabilities)?,
-                Identity::Anyone => {
-                    self.set_policy(DIEM_DEFAULT, engine, name, &perm.capabilities)?
-                }
-                Identity::NoOne => (),
-            };
-        }
-        Ok(())
-    }
-
     fn crypto_name(&self, name: &str) -> String {
-        self.name(name, &VaultEngine::Transit)
+        name.replace('/', "__")
```

Comment:
hmm... interesting, but now non-namespaced systems need to know about namespaces. I wonder if there's some cute way of making it expected that the CryptoStorage and KVStorage have a potentially specializable function namespace_separator to hide this? I know back in the day, Rust didn't support this kind of "overriding" of a default function implementation.

Reply:
I might have one idea to handle that. I can try that tomorrow when I'm back

Reply:
Yeah, it's not ideal that we'd need to expose everything to namespace separators, but I guess this is the trade-off we make.

Reply:
Yeah the thing is that this is a quirk of Vault so to get around it vault needs to do this special handling (which it really was already doing) or we can expose all storage impls to the concept of namespacing by adding a `namespace_separator` to both KVStorage and CryptoStorage traits (which feels worse now that i'm thinking about it)

Reply:
is the plan to revisit this?

Reply:
Since this is a problem with vaults storage it was cleaner to isolate the problem to the vault implementation rather than complicate all the other storage impls